base/src/xournalpp/gui/widgets/XournalWidget.cpp

```diff
@@ -0,0 +1,104 @@
+#include "XournalWidget.h"
+
+#include <cmath>
+#include <utility>
+
+#include <gdk/gdk.h>
+
+enum Direction { HORIZONTAL, VERTICAL };
+
+XournalWidget::XournalWidget(const lager::reader<Settings>& settings, lager::reader<Viewport> viewport,
```

Comment:
Why take `settings` by `const&` and `viewport` by value? I'd take both by value (they have reference semantics under the hood anyways as you know.)

Reply:
Oh, I see, because you are not copying `settings` into a local variable. You can still move `settings` into `operator[]` below though.